src/pixelator/graph/backends/implementations/_networkx.py

```diff
@@ -693,3 +705,69 @@ def subgraphs(self) -> Iterable[Graph]:
             Graph(NetworkXGraphBackend(self._graph.subgraph(cluster).copy()))
             for cluster in self._clustering
         ]
+
+
+def pmds_layout(
+    g: nx.Graph, pivots: int = 50, dim: int = 2, seed: Optional[int] = None
+) -> pd.DataFrame:
+    """Add some number of vertices to the graph instance.
+
+    :param g: A networkx graph object
+    :param pivots: The number of pivot nodes to use
+    :param dim: The dimension of the layout
+    :param seed: Set seed for reproducibility
+
+    :return: A dataframe with layout coordinates
+    :rtype: pd.DataFrame
+    :raises: ValueError raises if conditions are not met
+    """
+    if not isinstance(g, nx.Graph):
+        raise ValueError("Not a graph object")
+
+    if not nx.is_connected(g):
+        raise ValueError("Only connected graphs are supported.")
+
+    if pivots >= len(g.nodes):
+        raise ValueError("'pivots' must be less than the number of nodes in the graph.")
+    if pivots < dim:
+        raise ValueError(
+            "'pivots' must be greater than or equal to the dimension of the layout."
+        )
+    if pivots < 0:
```

Comment:
This is a bit weird in that the dimensions `dim` need to be already 2 or 3. Maybe we just need to review the order of the if statements.

Reply:
Based on our discussion this morning, I updated the check to this:
````
pivot_lower_bound = np.min([np.floor(0.2 * len(g.nodes)), 50])
    if pivots < pivot_lower_bound:
        raise ValueError(
            f"'pivots' must be greater than or equal to {pivot_lower_bound}"
        )
````
The number of pivot points are now expected to be at least floor(0.2*total_nodes). However, this number is capped at 50 so that larger graphs can still use from 50 pivot points (default) and upwards.